pkg/k8s/service_utils.go

```diff
@@ -50,3 +51,37 @@ func IsServiceHeadless(svc *corev1.Service) bool {
 	}
 	return false
 }
+
+// StripDownServiceTransformFunc is a transform function that strips down service to reduce memory usage.
+// see details in [stripDownServiceObject].
+func StripDownServiceTransformFunc(obj interface{}) (interface{}, error) {
+	if service, ok := obj.(*corev1.Service); ok {
+		return stripDownServiceObject(service), nil
+	}
+	return obj, nil
+}
+
+// stripDownServiceObject provides an stripDown version of service to reduce memory usage.
+// NOTE: if the controller needs to refer to more service fields in the future
+// these fields need to be added to the cache
+func stripDownServiceObject(service *corev1.Service) *corev1.Service {
+	service.ObjectMeta = metav1.ObjectMeta{
+		Name:              service.Name,
+		Namespace:         service.Namespace,
+		UID:               service.UID,
+		DeletionTimestamp: service.DeletionTimestamp,
+		ResourceVersion:   service.ResourceVersion,
+		Finalizers:        service.Finalizers,
```

Comment:
I guess the same question for services. Do we need `UID` and `Finalizers`?

Reply:
For `Finazlier` and `UID` I just keep them for safety, the field itself is small, just in case the controller needs it in the future.